malaria-app-android/src/main/java/com/peacecorps/malaria/ui/user_profile/UserProfilePresenter.java

```diff
@@ -0,0 +1,68 @@
+package com.peacecorps.malaria.ui.user_profile;
+
+import android.content.Context;
+import android.text.TextUtils;
+
+import com.peacecorps.malaria.data.AppDataManager;
+import com.peacecorps.malaria.ui.base.BasePresenter;
+import com.peacecorps.malaria.ui.user_profile.UserProfileContract.UserProfileMvpPresenter;
+import com.peacecorps.malaria.ui.user_profile.UserProfileContract.UserProfileMvpView;
+
+/**
+ * Created by Anamika Tripathi on 13/7/18.
+ */
+public class UserProfilePresenter<V extends UserProfileMvpView> extends BasePresenter<V> implements UserProfileMvpPresenter<V> {
+
+    UserProfilePresenter(AppDataManager manager, Context context) {
+        super(manager, context);
+    }
+
+    @Override
+    public void setPreviousDetails() {
+        getView().setInitialValuesIfAvailable(
+                getDataManager().getUserName(),
+                getDataManager().getUserEmail(),
+                getDataManager().getUserAge(),
+                getDataManager().getDrugPicked()
+        );
+    }
+    @Override
+    public void attachView(V view) {
+        super.attachView(view);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+    }
+
+    @Override
+    public void setNewDetails(String name, String email, int age) {
+        getDataManager().setUserName(name);
+        getDataManager().setUserEmail(email);
+        getDataManager().setUserAge(age);
+    }
+
+    @Override
+    public boolean testIsEmpty(String text) {
+        return TextUtils.isEmpty(text);
+    }
+
+    @Override
+    public boolean isAgeValid() {
+        if(testIsEmpty(getView().getUserAge()))
+            return false;
+        String text = getView().getUserAge();
```

Comment:
This doesn't look right to me, you are checking Age validity, with String length for that you are converting to String. Instead, I would recommend to have check, something like this age>0 && age <=100 (looks like you are having upper bound on age).

Reply:
Thank you. Improved it locally & will include in next PR. Also, I've added comments now throughout user_profile directory, sorry for confusion above.